File: app/src/main/java/org/mifos/mobile/ui/views/AccountsView.java
package org.mifos.mobile.ui.views;

import org.mifos.mobile.models.accounts.loan.LoanAccount;
import org.mifos.mobile.models.accounts.savings.SavingAccount;
import org.mifos.mobile.models.accounts.share.ShareAccount;
import org.mifos.mobile.ui.views.base.MVPView;

import java.util.List;

/**
 * Created by Rajan Maurya on 23/10/16.
 */

public interface AccountsView extends MVPView {

    void showLoanAccounts(List<LoanAccount> loanAccounts);

    void showSavingsAccounts(List<SavingAccount> savingAccounts);

    void showShareAccounts(List<ShareAccount> shareAccounts);

    void showError(String errorMessage);
}

File: app/src/main/java/org/mifos/mobile/ui/views/AccountsView.kt
package org.mifos.mobile.ui.views

import org.mifos.mobile.models.accounts.loan.LoanAccount
import org.mifos.mobile.models.accounts.savings.SavingAccount
import org.mifos.mobile.models.accounts.share.ShareAccount
import org.mifos.mobile.ui.views.base.MVPView

/**
 * Created by Rajan Maurya on 23/10/16.
 */
interface AccountsView : MVPView {
    fun showLoanAccounts(loanAccounts: List<LoanAccount?>?)
    fun showSavingsAccounts(savingAccounts: List<SavingAccount?>?)
    fun showShareAccounts(shareAccounts: List<ShareAccount?>?)
    fun showError(errorMessage: String?)
}